menas/src/main/scala/za/co/absa/enceladus/menas/services/VersionedModelService.scala

```diff
@@ -70,16 +70,28 @@ abstract class VersionedModelService[C <: VersionedModel with Product with Audit
     for {
       versions <- {
         //store all in version ascending order
-        val all = versionedMongoRepository.getAllVersions(name, true).map(_.sortBy(_.version))
+        val all = versionedMongoRepository.getAllVersions(name, inclDisabled = true).map(_.sortBy(_.version))
         //get those relevant to us
-        if (fromVersion.isDefined) all.map(_.filter(_.version <= fromVersion.get)) else all
+        if (fromVersion.isDefined) {
+          all.map(_.filter(_.version <= fromVersion.get))
+        } else {
+          all
+        }
       }
       res <- {
         //see if this was branched from a different entity
-        val topParent = if (versions.isEmpty) None
-          else if (versions.head.parent.isEmpty) None
-          else versions.head.parent
-        if(topParent.isDefined) getParents(topParent.get.name, Some(topParent.get.version)) else Future.successful(Seq())
+        val topParent = if (versions.isEmpty) {
+          None
+        } else if (versions.head.parent.isEmpty) {
```

Comment:
These few lines can probably be combined to one clause:
```suggestion
        if (versions.isEmpty || versions.head.parent.isEmpty) {
```